model-optimizer/extensions/front/tf/KerasRNNReplacer.py

```diff
@@ -0,0 +1,237 @@
+"""
+ Copyright (C) 2017-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.front.tf.WhileNormalize import WhileNormalize
+from extensions.ops.loop import Loop
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.pattern_match import find_pattern_matches, inverse_dict
+from mo.ops.const import Const
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+def find_intergraph_match(graph: Graph, ext_match: dict, reserve_port: int, unstack_port: int, stack_port: int):
+    """
+    The function checks that logic using TensorList operations corresponds to subsequent slicing of input and
+    concatenation of intermediate results for output.
+    If pattern is successfully found, it returns matching dictionary.
+    """
+    def back_edge_exists(back_edges_map, from_layer, to_layer, from_port=0, to_port=0):
+        for back_edge in back_edges_map:
+            if back_edge['from_layer'] == from_layer and back_edge['to_layer'] == to_layer \
+                    and back_edge['from_port'] == from_port and back_edge['to_port'] == to_port:
+                return True
+        return False
+
+    def inter_edge_exists(input_port_map, external_port_id, internal_layer_id):
```

Comment:
`inner` or `intra` instead of `inter`?

Reply:
Data types

Reply:
it is neither `inner` nor `intra`. It connects external input port of Loop node and internal input port of a body.